tests/unit/test_link_generation_facade.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# ======================================================================================================================
+# Imports
+# ======================================================================================================================
+from zigzag.link_generation_facade import LinkGenerationFacade
+
+ASC_TESTSUITE_PROPS = {
+    'REPO_URL': 'https://github.com/rcbops/rpc-openstack',
+    'RE_JOB_BRANCH': 'pike-rc',
+    'MOLECULE_TEST_REPO': 'molecule-validate-neutron-deploy',
+    'MOLECULE_SCENARIO_NAME': 'default',
+}
+
+MK8S_TESTSUITE_PROPS = {
+    'GIT_URL': 'https://github.com/rcbops/mk8s.git',
+    'GIT_BRANCH': 'master',
+}
+
+LINE_NUMBER = '42'
+
+FAILURE_OUTPUT = """
+    @pytest.mark.test_id('d7fc646b-432a-11e8-b858-6a00035510c0')
+    @pytest.mark.jira('asc-239')
+    def test_verify_network_list(host):
+        \"\"\"Verify the neutron network was created\"\"\"
+        cmd = pre_cmd + "openstack network list\""
+        output = host.run(cmd)
+&gt;       assert ("GATEWAY_NET" in output.stdout)
+E       assert 'GATEWAY_NET' in ''
+E        +  where '' = CommandResult(command='bash -c "source /root/openrc; openstack network list"', exit_status=0, stdout=u'', stderr=u'').stdout
+
+tests/test_verify_networks_created.py:23: AssertionError
+""".format(LINE_NUMBER)  # noqa
+
+TEST_FILE = 'tests/totally_real/verify_I_am_good_link.py'
+
+
+# ======================================================================================================================
+# Test Suites
+# ======================================================================================================================
+class TestLinkGenerationFacade(object):
+    """Tests for the LinkGenerationFacade"""
+
+    def test_asc_failure_link(self, mocker):
+        """Validate when configured with asc as ci-environment"""
+        zz = mocker.MagicMock()
+        zz.ci_environment = 'asc'
+        zz.testsuite_props = ASC_TESTSUITE_PROPS
+
+        zztl = mocker.MagicMock()
+        zztl.test_file = TEST_FILE
+        zztl.failure_output = FAILURE_OUTPUT
+
+        lgf = LinkGenerationFacade(zz)
+        failure_link = lgf.github_testlog_failure_link(zztl)
+        assert failure_link == 'https://github.com/rcbops/molecule-validate-neutron-deploy/blob/pike-rc/molecule/default/tests/totally_real/verify_I_am_good_link.py#L23'  # noqa
```

Comment:
The validation string here is too magical.

I do not know how the following values were derived:
* The github namespace expected for the `MOLECULE_TEST_REPO` defined in `ASC_TESTSUITE_PROPS`: `rcbops`.
* The molecule scenario name under test: `default`

The following value can be discovered, but is a bit obscure:
* The line number associated with the failure in the test: `23`.